x/interchainqueries/keeper/keeper.go

```diff
@@ -334,3 +370,55 @@ func (k Keeper) MustPayOutDeposit(ctx sdk.Context, deposit sdk.Coins, sender sdk
 		panic(err.Error())
 	}
 }
+
+// GetTxQueriesToRemove retrieves the list of TX queries registered to be removed. Returns a slice
+// with no more than limit entities or all entities if limit is 0.
+func (k Keeper) GetTxQueriesToRemove(ctx sdk.Context, limit uint64) []uint64 {
+	prefixStore := prefix.NewStore(ctx.KVStore(k.storeKey), types.TxQueryToRemoveKey)
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+	ids := make([]uint64, 0, 100)
+	for ; iterator.Valid(); iterator.Next() {
+		ids = append(ids, sdk.BigEndianToUint64(iterator.Key()))
+		if limit != 0 && uint64(len(ids)) >= limit {
+			return ids
+		}
+	}
+	if len(ids) == 0 {
+		return nil
+	}
+	return ids
+}
+
+// getTxQueryHashes retrieves up to limit tx hashes related to a query with the given ID from the
+// Keeper's store and a bool meaning if all the query's hashes have been retrieved. If limit is 0,
+// it retrieves all the hashes for the given query.
+func (k Keeper) getTxQueryHashes(ctx sdk.Context, queryID uint64, limit uint64) (txHashes [][]byte, complete bool) {
```

Comment:
Pretty weird method for me, actually.
Why not to return `TxQueryToRemove` structure itself?

Reply:
good idea!

Reply:
done. now it's much more clear, ty